bundles/framework/myplaces3/MyPlacesLayerForm.jsx

```diff
@@ -1,46 +1,74 @@
 
 import { StyleEditor } from 'oskari-ui/components/StyleEditor';
 import { OSKARI_BLANK_STYLE } from 'oskari-ui/components/StyleEditor/index';
+import { LOCALE_KEY } from './constants';
 
-export const MyPlacesLayerForm = ({ name, style, onSave, onCancel }) => {
+const PaddedInput = styled(TextInput)`
+    margin-bottom: 10px;
+`;
+
+const getPlaceholders = () => {
+    const localizedFields = { name: Oskari.getMsg(LOCALE_KEY, 'categoryform.layerName') };
+    return createLocalizedLabels(localizedFields);
+};
+
+const getMandatory = (defaultLang) => {
+    const mandatory = {};
+    mandatory[defaultLang] = ['name'];
+    return mandatory;
+};
+
+// name in default language is mandatory
+const validate = (lang, value) => lang !== Oskari.getDefaultLanguage() || (typeof value === 'string' && value.trim().length > 0);
+
+export const MyPlacesLayerForm = ({ locale: initLocale, style: initStyle, onSave, onCancel }) => {
     const [editorState, setEditorState] = useState({
-        style: style || OSKARI_BLANK_STYLE,
-        name: name
+        style: initStyle || OSKARI_BLANK_STYLE,
+        locale: initLocale || {}
     });
-
+    const { locale, style } = editorState;
     const updateStyle = (style) => setEditorState({ ...editorState, style });
-    const updateName = (name) => setEditorState({ ...editorState, name });
-    const hasName = editorState.name.length > 0;
+    const updateLocale = (locale) => setEditorState({ ...editorState, locale });
+    const defaultLang = Oskari.getDefaultLanguage();
+    const hasName = Oskari.util.keyExists(locale, `${defaultLang}.name`) && locale[defaultLang].name.trim().length > 0;
+    // TODO: show warning in button tooltip
+    // { !hasName && <Message messageKey='validation.categoryName' /> }
     return (
         <Modal
             title={ <Message messageKey={ 'categoryform.title' } /> }
             visible={ true }
-            onOk={ () => onSave(editorState.name, editorState.style) }
+            onOk={ () => onSave(locale, style) }
+            maskClosable = { false }
             okButtonProps={ { disabled: !hasName } }
             onCancel={ onCancel }
             cancelText={ <Message messageKey="buttons.cancel" /> }
             okText={ <Message messageKey="buttons.save" /> }
         >
-            <TextInput
-                addonBefore={ <Message messageKey={ 'categoryform.layerName' } /> }
-                value={ editorState.name }
-                onChange={ (event) => updateName(event.target.value) }
-            />
-            { !hasName && <Message messageKey='validation.categoryName' /> }
+            <LocalizationComponent
+                placeholders={ getPlaceholders() }
+                value={ locale }
+                languages={ Oskari.getSupportedLanguages() }
+                onChange={ updateLocale }
+                mandatoryFields = { getMandatory(defaultLang) }
+            >
+                <PaddedInput type='text' name='name' validate={validate}/>
```

Comment:
Let's try passing placeholder with the input `<PaddedInput placeholder={'placeholder text'} useLangSuffix={true} />` instead of a prop object for LocalizationComponent. In the above example useLangSuffix=true would make LocalizationComponent attach the "in english" etc to the placeholder text value and false would leave it as is. Maybe default it to true even?

Reply:
Maybe we should just leave out the suffix prop for placeholder and assume it is always wanted. If we need to filter it out for some future case we can add it then.